sw/device/silicon_creator/rom/e2e/BUILD

```diff
@@ -854,3 +857,208 @@ test_suite(
+
+        echo :::: Configure the PMP unit.\\n
+        monitor reg pmpcfg0
+        # Write "L NAPOT X W R" to pmp{0,1,2,3}cfg in pmpcfg0. Crucially, this
+        # value is no less permissive than whatever the current value is.
+        monitor reg pmpcfg0 0x9f9f9f9f
+        monitor reg pmpaddr0 0xffffffff
+
+        echo :::: Value of CREATOR_SW_CFG_ROM_EXEC_EN.\\n
+        monitor mdw 0x40131108
+
+        echo :::: Load the SRAM program onto the device and check integrity.\\n
+        file sram_program.elf
+        load sram_program.elf
+        compare-sections
+
+        echo :::: Update registers before calling functions.\\n
+        set $sp = _stack_end
+        set $gp = __global_pointer$
+        info registers
+
+        # When testing SRAM execution, we want to be sure the code is running
+        # out of SRAM and not the instruction cache.
+        echo :::: Invalidate the icache.\\n
+        print icache_invalidate()
+
+        echo :::: Call sram_main().\\n
+        print sram_main()
+
+        echo :::: Done.\\n
+"""
+
+[
+    bitstream_splice(
+        name = "rom_otp_{}_exec_disabled".format(otp_name),
+        src = "//hw/bitstream:rom",
+        data = "img_{}_exec_disabled".format(otp_name),
+        meminfo = "//hw/bitstream:otp_mmi",
+        tags = [
+            "manual",
+            "vivado",
+        ],
+        update_usr_access = True,
+        visibility = ["//visibility:private"],
+    )
+    for otp_name in OTP_IMAGES
+]
+
+[
+    opentitan_gdb_fpga_cw310_test(
+        name = "sram_program_fpga_cw310_test_otp_" + otp_name,
+        timeout = "short",
+        exit_success_pattern = "sram_program\\.c:\\d+\\] PC: 0x100020e0, SRAM: \\[0x10000000, 0x10020000\\)",
+        gdb_script = SRAM_JTAG_INJECTION_GDB_SCRIPT,
+        gdb_script_symlinks = {
+            "//sw/device/examples/sram_program:sram_program_fpga_cw310.elf": "sram_program.elf",
+        },
+        rom_bitstream = ":rom_otp_{}_exec_disabled".format(otp_name),
+        rom_kind = "Rom",
+        tags = [
+            "cw310",
+            "vivado",
+        ],
+    )
+    for otp_name in OTP_IMAGES
+]
+
+test_suite(
+    name = "rom_e2e_jtag_inject_tests",
+    tags = [
+        "cw310",
+        "vivado",
+    ],
+    tests = ["sram_program_fpga_cw310_test_otp_" + otp_name for otp_name in OTP_IMAGES],
+)
+
+[
+    opentitan_gdb_fpga_cw310_test(
+        name = "asm_interrupt_handler_fpga_cw310_test_otp_" + otp_name,
+        timeout = "short",
+        gdb_script = """
+            target extended-remote :3333
+
+            echo :::: Send OpenOCD the 'reset halt' command.\\n
+            monitor reset halt
+
+            echo :::: Load ROM symbols into GDB.\\n
+            file rom.elf
+
+            echo :::: Set breakpoint on exception handler.\\n
+            break _asm_exception_handler
+
+            echo :::: Attempt to trigger an exception.\\n
+            set $pc = 0
```

Comment:
The test plan actually said to set PC to the end of ROM. @alphan Was that detail important, or is 0x0 just as good? I just arbitrarily chose zero while writing this because it was easier to calculate — I suppose I should double check that it's unmapped.

Reply:
This should be fine, as long as the address triggers an exception we are consistent with the test point description.